test/uk/gov/hmrc/tai/service/TaxAccountSummaryServiceSpec.scala

```diff
@@ -0,0 +1,422 @@
+import mocks.MockTemplateRenderer
+import org.joda.time.LocalDate
+import org.jsoup.Jsoup
+import org.mockito.{Matchers, Mockito}
+import org.mockito.Matchers.any
+import org.mockito.Mockito.when
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.mockito.MockitoSugar
+import org.scalatestplus.play.PlaySpec
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.test.Helpers.{contentAsString, status, _}
+import uk.gov.hmrc.domain.{Generator, Nino}
+import uk.gov.hmrc.http.{BadRequestException, HeaderCarrier}
+import uk.gov.hmrc.play.audit.http.connector.AuditConnector
+import uk.gov.hmrc.play.frontend.auth.connectors.{AuthConnector, DelegationConnector}
+import uk.gov.hmrc.play.partials.FormPartialRetriever
+import uk.gov.hmrc.tai.connectors.responses.{TaiSuccessResponseWithPayload, TaiTaxAccountFailureResponse}
+import uk.gov.hmrc.tai.model.TaxYear
+import uk.gov.hmrc.tai.model.domain._
+import uk.gov.hmrc.tai.model.domain.income._
+import uk.gov.hmrc.tai.util.TaxYearRangeUtil
+import uk.gov.hmrc.tai.util.constants.AuditConstants
+import uk.gov.hmrc.tai.viewModels.{IncomesSources, TaxAccountSummaryViewModel}
+
+import scala.concurrent.duration._
+import scala.concurrent.{Await, Future}
+import scala.util.Random
+
+class TaxAccountSummaryServiceSpec extends PlaySpec
+  with MockitoSugar
+  with FakeTaiPlayApplication
+  with I18nSupport
+  with AuditConstants
+  with BeforeAndAfterEach {
+
+  implicit val messagesApi: MessagesApi = app.injector.instanceOf[MessagesApi]
+  override def beforeEach: Unit = {
+    Mockito.reset(taxAccountService)
+    Mockito.reset(employmentService)
+  }
+
+
+  "TaxAccountSummaryServiceSpec" should {
+    "return a BadRequestException if ceasedEmployments fails" in {
+      val sut = createSUT
+      when(employmentService.ceasedEmployments(any[Nino], any[TaxYear])(any[HeaderCarrier])).thenReturn(
+        Future.failed(new BadRequestException("Failed Call"))
+      )
+
+      when(taxAccountService.incomeSources(any[Nino], any[TaxYear], Matchers.eq(PensionIncome), Matchers.eq(Live))(any[HeaderCarrier])).thenReturn(
+        Future.successful(TaiSuccessResponseWithPayload[Seq[IncomeSource]](Seq.empty[IncomeSource]))
+      )
+
+      when(taxAccountService.incomeSources(any[Nino], any[TaxYear], Matchers.eq(EmploymentIncome), Matchers.eq(Live))(any[HeaderCarrier])).thenReturn(
+        Future.successful(TaiSuccessResponseWithPayload[Seq[IncomeSource]](Seq.empty[IncomeSource]))
+      )
+
+      when(taxAccountService.incomeSources(any[Nino], any[TaxYear], Matchers.eq(EmploymentIncome), Matchers.eq(Ceased))(any[HeaderCarrier])).thenReturn(
+        Future.successful(TaiSuccessResponseWithPayload[Seq[IncomeSource]](Seq.empty[IncomeSource]))
+      )
+
+      when(taxAccountService.nonTaxCodeIncomes(any(), any())(any())).thenReturn(
+        Future.successful(TaiSuccessResponseWithPayload[NonTaxCodeIncome](nonTaxCodeIncome))
+      )
+
```

Comment:
These mock overrides are really hard to read and to see what is happening because they are in each function and very long. To simplify each test, you can place the "happy path" (the ones returning emptys) into the SUT's constructor, this will allow you to deduplicate lines 70-85 and just have the failing mock override call in the test. It will make it clear what your intentions are for each test case as you are overriding default (happy path) behaviour

Reply:
agreed, updated